### PR TITLE
Remove matrix and matrices

### PR DESCRIPTION
**Pull request by jl-wynen**
Long overdue. I found no usages in scippneutron or ess.

### CODE REVIEW

File: tests/factory.py [hunk 1]
```diff
@@ -169,7 +169,6 @@ def make_simple_datagroup(child=None, maxdepth=1, cur_depth=1):
         dims=['x'], values=[[1, 2, 3], [1, 2, 3], [1, 2, 3]]
     )
     dg['trans3-' + id_suffix] = trans3
-    dg['matrix-' + id_suffix] = sc.matrix(value=[[1, 2.5, 3.75], [1, 2, 3], [1, 2, 3]])
```
[the remark this answers is not in the document]
Comment (jl-wynen) [reply tests/factory.py:172]: What do you mean? This dg entry should have been used but can no longer be.

File: tests/linalg_test.py [hunk 2]
```diff
@@ -123,90 +123,6 @@ def test_matrix_from_quat_coeffs_numpy():
...
-
-def test_variable_1D_matrix_from_numpy():
-    data = np.array(
-        [
-            np.arange(9.0).reshape(3, 3),
-            np.arange(5.0, 14.0).reshape(3, 3),
-            np.arange(1.0, 10.0).reshape(3, 3),
-        ]
-    )
-    var = sc.matrices(dims=['x'], values=data, unit=sc.units.us)
```
[the remark this answers is not in the document]
Comment (jl-wynen) [reply tests/linalg_test.py:152]: Good point. I was under the assumption that there are tests for linear transform. But there aren't.

File: tests/factory.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply tests/factory.py:172]: Matrices are now called linear transforms, but it is the same otherwise?

[the remark this answers is not in the document]
Comment (jl-wynen) [reply tests/factory.py:172]: Added a linear transform.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**